web/web-sockets-worker.js

```diff
@@ -0,0 +1,72 @@
+var webSocket;
+const broadcast = new BroadcastChannel("background-message");
+var intervalId;
+const pingIntervalInMs = 20000;
+
+function connect(url, ticket) {
+  webSocket = new WebSocket(`${url}?ticket=${ticket}`, "jmap");
+
+  webSocket.onopen = () => {
+    console.log("websocket open");
+    webSocket.send(
+      JSON.stringify({
+        "@type": "WebSocketPushEnable",
+        dataTypes: ["Mailbox", "Email", "EmailDelivery"],
+      })
+    );
+    intervalId = setInterval(() => {
+      webSocket.send(
+        JSON.stringify({
+          "@type": "Request",
+          id: "R1",
+          using: ["urn:ietf:params:jmap:core"],
+          methodCalls: [["Core/echo", {}, "c0"]],
```

Comment:
what is this?

Reply:
The ping request, to keep the connection alive.